pyfranca/franca_lexer.py

```diff
@@ -118,22 +124,47 @@ def t_STRUCTURED_COMMENT(t):
         r"<\*\*(.|\n)*?\*\*>"
         t.lexer.lineno += t.value.count("\n")
 
-    # Identifier
+    # File name (quoted)
     # noinspection PyPep8Naming,PyIncorrectDocstring
     @staticmethod
-    def t_ID(t):
+    def t_FILE_NAME(t):
         # noinspection PySingleQuotedDocstring
-        r"[A-Za-z][A-Za-z0-9_]*"
-        t.type = Lexer._keyword_map.get(t.value, "ID")
+        r"\"([^\n]|\.)*?\""
+        t.value = t.value[1:-1]
         return t
 
-    # File name (quoted)
     # noinspection PyPep8Naming,PyIncorrectDocstring
     @staticmethod
-    def t_FILE_NAME(t):
+    def t_STRING_VAL(t):
         # noinspection PySingleQuotedDocstring
-        r"\"([^\n]|\.)*?\""
-        t.value = t.value[1:-1]
+        r"\"[^\"]*\""
+        t.value = t.value.replace("\"", '')
+        t.value = str(t.value)
+        return t
+
+    # noinspection PyPep8Naming,PyIncorrectDocstring
+    @staticmethod
+    def t_FLOAT_VAL(t):
+        # noinspection PySingleQuotedDocstring
+        # r"[-+]?(\d+)?(\.(\d+)?([eE][-+]?\d+)?|[eE][-+]?\d+)[f]"
+        # currently valid float string without leading digits (.043f) are not supported
+        # result in conflict with text like org.franca... matche ".f" --> need to improve regex
```

Comment:
I am concerned about this ambiguity... I would like to understand where this ambiguity comes from before merging as the lexer and parser should be perfectly capable of dealing with this. Maybe some of my FQN grammar rules are not good enough?